cometsdk.go

```diff
@@ -4113,6 +4133,8 @@ func (this *CometAPIClient) AdminAccountSessionStartAsUser(TargetUser string) (*
 	if err != nil {
 		return nil, err
 	}
+	// Set the SessionKey so that future invocations use the SessionKey.
+	c.SessionKey = result.SessionKey
```

Comment:
I'm aware of at least one SDK user who uses this `AdminAccountSessionStartAsUser` function to generate a SessionKey for a customer, and then hands it off to the browser for SSO (as per https://docs.cometbackup.com/latest/api/other-apis#usersessionlogin).

This is a breaking change for them, since the API client would no longer function after one call to this function.

For the other SessionLogin family of functions, though, it is a nice improvement, primarily for `AdminAccountSessionStart`.

- Can you restrict the change to just AdminAccountSessionStart? Or perhaps, only do the AuthType transform in AdminAccountSessionStart if AuthType is already != "Password"?
- Maybe the caller should always have to manually change the AuthType themselves

Reply:
Thanks for that context on the backward compatibility considerations, I missed that.

In trying to make the API convenient to use and more fail-safe, I propose adding another option in the `CometAPIClient` called `ReuseSessionKey`, which defaults to `false`.
When the caller is setting the `TOTPKey`, they are effectively already indicating that they would like to use `AuthType` `"PasswordTOTP"`. So, expected them to set `AuthType` as well may be redundant and could also lead to misconfigurations/more involved configuration steps.
Similarly, when they set `SessionKey` manually or automatically, it would be reasonable to assume that they want to authenticate using `AuthType` `"SessionKey"`.
Handling that logic internally while explaining the usage clearly in the documentation may be more convenient to use.
As you rightly pointed out the SessionKey login is more useful for AdminAccountSessionStart. There may be a few cases where it could be useful for some of the other SessionStart apis as well.
By using the ReuseSessionKey option, we let the caller decide whether they want to use it or not, while hiding all the details within the SDK implementation. This way it's probably more flexible in supporting the various use cases.

I've added more detailed documentation on the various options as well.

Feel free to make updates as you see fit and I would be glad to make any further modifications that are needed as well.

Reply:
Also, I haven't added support for PasswordU2F as I haven't used it yet, but I imagine it may be simple to incorporate along the same lines to make this a more wholesome update.